Makefile.generated_files

```diff
@@ -12,6 +12,12 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Don't allow users to call this directly.  There are too many variables this
+# assumes to inherit from the main Makefile.  This is not a user-facing file.
+ifeq ($(CALLED_FROM_MAIN_MAKEFILE),)
+    $(error Please use the main Makefile to access these rules)
```

Comment:
I would put a little more information here in message or in the comment to run "make generated_files" rule. I assume that is the only way to run rules in this file.